airflow-core/src/airflow/api_fastapi/common/parameters.py

```diff
@@ -221,9 +246,13 @@ def get_primary_key_string(self) -> str:
     def depends(cls, *args: Any, **kwargs: Any) -> Self:
         raise NotImplementedError("Use dynamic_depends, depends not implemented.")
 
-    def dynamic_depends(self, default: str | None = None) -> Callable:
-        def inner(order_by: str = default or self.get_primary_key_string()) -> SortParam:
-            return self.set_value(self.get_primary_key_string() if order_by == "" else order_by)
+    def dynamic_depends(self, default: list[str] | None = None) -> Callable:
+        def inner(order_by: list[str] | str | None = Query(default)) -> SortParam:
+            if order_by is None:
+                order_by = [self.get_primary_key_string()]
+            elif isinstance(order_by, str):
+                order_by = [order_by]
```

Comment:
This shouldn't be possible. FastAPI validation will prevent such wrong input and will do what's necessary to either get a `list[str]` or fail with a 422 validation error if not possible.